code/core_allvars.h

```diff
@@ -278,5 +294,27 @@ extern gsl_rng *random_generator;
 extern int TreeID;
 extern int FileNum;
 
+#ifdef HDF5
+extern char          *core_output_file;
+extern size_t         HDF5_dst_size;
+extern size_t        *HDF5_dst_offsets;
+extern size_t        *HDF5_dst_sizes;
+extern const char   **HDF5_field_names;
+extern hid_t         *HDF5_field_types;
+extern int            HDF5_n_props;
+#endif
+
+
+#define DOUBLE 1
+#define STRING 2
+#define INT 3
+
+enum Valid_TreeTypes
+{
+  genesis_lhalo_hdf5 = 0,
+  lhalo_binary = 1
```

Comment:
You can add a named field, something like `num_tree_types` field as the last enum, like so:

```C
enum Valid_TreeTypes
{
  genesis_lhalo_hdf5 = 0,
  lhalo_binary = 1,
  num_tree_types
}
```
Then, the `num_tree_types` field will auto-increment the last enum value by 1, i.e., `num_tree_types := 2`.
Afterwards, you can write input validation code like this:

```C
if(tree_type < 0 || tree_type >= num_tree_types) {
    /* print error message */
}
```

Reply:
This causes a warning in my switch statements when I decide what tree reader to use ``code/core_io_tree.c:73: warning: enumeration value 'num_tree_types' not handled in switch``.  

I could obviously include the case explicitly in the switch but this seems redundant; I validate the user's value of ``TreeType`` in ``read_parameter_file()`` which is called before the switch.  There's no point in checking for ``case(num_tree_types)`` because we will error-exit before we hit the switch!